test/tests/parallel/set3/mta/smtp_delivery.bats

```diff
@@ -49,7 +51,11 @@ function setup_file() {
   assert_success
   _wait_until_change_detection_event_completes
 
-  _wait_for_smtp_port_in_container
+  # Even if the Amavis port is reachable at this point, it may still refuse connections?
+  _wait_for_tcp_port_in_container 10024
```

Comment:
I'm not actually sure when these are used :sweat_smile: 

Perhaps I'm listening on the wrong port since 10024 appears to be for sending and 10025 for receiving? I just know that when I queried the mail queue, sometimes a connection refused error mentioned port 10024.

> You could do me a favour and run the tests locally a few times so we can be more sure about not having introduced a new race condition here.

I ran it many times actually and wasn't able to produce it locally. I had this change ready around a month ago, but back then it occurred more often. I had a solution to delay sending mail a little bit with `sleep 1`:

```bash
  # NOTE: If change detection begins too early, it can cause connection issues with Amavis.
  # `sleep 1` can be added before or after to prevent that from happening.
  sleep 1
```

I am pretty sure that the issue is more to do with Amavis not being ready quick enough, as before with the 10 sec change event delay, there was plenty of time. There might be a small window after the change event with the reloads on amavis and postfix (plus dovecot for delivery I guess).

---

I'll be getting some rest, but will address this concern when I'm back :+1:

Reply:
> I'll be getting some rest, but will address this concern when I'm back 👍

Alright! I too think this is about a Amavis-timing-issue. I'd add the check for port 10025 and I'm not against a small sleep as mentioned in your comment (with additional info on why it's needed, just like in your comment).

Reply:
>  I'd add the check for port 10025

Will this be added or should I approve now?

Reply:
I'll update it, sorry been a bit busy and since CI is broken presently for tests it seems I should wait until we've resolved that :sweat_smile:

Reply:
> I'd add the check for port 10025

Will this be added or should I approve now? 😆

Reply:
Oh sorry, I don't believe there needs to be a check for port 10025. AFAIK this is a port on Postfix to receive mail back from Amavis after it's done processing. I am doubtful it'll make a difference when there's a connection error logged about port 10024. The sleep should give Amavis enough time.